src/platform_impl/macos/util/async.rs

```diff
@@ -251,14 +243,9 @@ pub unsafe fn set_focus(ns_window: id) {
 // through the `IdRef` because otherwise it would dereference free'd memory
 pub unsafe fn close_async(ns_window: IdRef) {
   let ns_window = MainThreadSafe(ns_window);
-  Queue::main().exec_async(move || {
+  run_on_main(move || {
     autoreleasepool(move || {
       ns_window.close();
-      let event = Event::WindowEvent {
-        window_id: WindowId(get_window_id(*ns_window.0)),
-        event: WindowEvent::Destroyed,
```

Comment:
why was this event removed? is the destroyed event sent somewhere else?

Reply:
This is a workaround we did before.
It should be sent by `windowWillClose` in window's delegate.